internal/armada/server/submit_from_log.go

```diff
@@ -47,7 +47,10 @@ func (srv *SubmitFromLog) Run(ctx context.Context) error {
 		if err := recover(); err != nil {
 			log.WithField("error", err).Error("unexpected panic; restarting")
 			time.Sleep(time.Second)
-			go srv.Run(ctx)
+			go func() {
+				err := srv.Run(ctx)
+				log.WithField("error", err).Error("unexpected panic restarting again")
```

Comment:
Bug.  And this could become a recursive call if run keeps failing.

Reply:
what's the bug? it doesn't look potentially recursive, since the go func is not calling panic()

Reply:
srv.Run() returning an error is not a panic

Reply:
The bug is on my end but its fixed.  I log an error no matter if err is nil or not.  I did that in a few places so bug is mostly a note to me to fix that.  

The recursive call is that this function is already srv.Run so we are recovering from a panic and then calling the same function again.  If we keep failing then it could be a problem.

Reply:
maybe we shouldn't recover from panic with restart? or some sort backoff on restarts by counting (?)